scene/gui/text_edit.cpp

```diff
@@ -5860,3 +5757,216 @@ TextEdit::TextEdit() {
 
 TextEdit::~TextEdit() {
 }
+
+///////////////////////////////////////////////////////////////////////////////
+
+Map<int, TextEdit::HighlighterInfo> TextEdit::_get_line_syntax_highlighting(int p_line) {
+	if (syntax_highlighter != NULL) {
+		return syntax_highlighter->_get_line_syntax_highlighting(p_line);
+	}
+
+	Map<int, HighlighterInfo> color_map;
+
+	bool prev_is_char = false;
+	bool prev_is_number = false;
+	bool in_keyword = false;
+	bool in_word = false;
+	bool in_function_name = false;
+	bool in_member_variable = false;
+	bool is_hex_notation = false;
+	Color keyword_color;
+	Color color;
+
+	int in_region = -1;
+	int deregion = 0;
+	for (int i = 0; i < p_line; i++) {
```

Comment:
It seems that going through the whole text for getting a line color is pretty inefficient. There's probably a better way to handle multiple line coloring.

Reply:
I think doing this is fine, but a caching system with proper invalidation should be used instead. Ie, if you edit line 5, the highlighting for all lines below 5 becomes invalid.

Reply:
A line cache should simply remember the hlighliter computed status you already have in that line, and the last valid line, so you just run from that line onwards when redrawing. Should be very efficient even for very large files.

Reply:
I guess great minds think alike ;)

I have some changes locally around the idea of caching then dropping everything below and including the edited line. Was going to send that in a separate PR when I can get some more time to finish it. 

However, yeah at the moment it's not the most efficient solution, but should be good enough.